### PR TITLE
Formatting settings same for both `type` and `var` typeexpr

### PR DESCRIPTION
**Pull request by piyush0411**
Signed-off-by: piyush0411 <piyush04111999@gmail.com>

### CODE REVIEW

**Comment by piyush0411** [comment]
@lydia-duncan - These were initial changes that might solve #17414 . Though I am not able to figure out the corner cases, as you discussed in the issue comments.
Kindly help me with these cases.

Thanks :)

**Comment by piyush0411** [comment]
I tried testing these changes with a small test that produces the following output
```
.. default-domain:: chpl

.. module:: functionTypeexprFormat.doc

functionTypeexprFormat.doc
==========================
**Usage**

.. code-block:: chapel

   use functionTypeexprFormat.doc;


or

.. code-block:: chapel

   import functionTypeexprFormat.doc;

.. type:: type t = 2*(int)

.. data:: var x: 2*(int)

.. data:: param w = 32

.. type:: type t2 = int(w/2)

.. data:: var x2: int(w/2)

```

So I guess the changes work for `type`.

**Comment by lydia-duncan** [comment]
Since the code from the original issue was not already saved as a test, I won't accept this PR without committed test cases.

**Comment by lydia-duncan** [comment]
> Though I am not able to figure out the corner cases, as you discussed in the issue comments.

I'm sorry, but I need you to try to put more effort into coming up with cases.  It currently feels like you're frequently waiting for me to direct you, and that's not something I will have time for on a regular basis, especially not at this level of granularity - otherwise I would have solved these issues myself already.  But in addition to the amount of time I'm spending, I also want you to be successful - you'll have a harder time finding other work if you aren't solving problems independently and coming up with test cases on your own.  I've been trying to encourage this subtly but it felt like I needed to be more direct to make it clear.  This doesn't mean I won't help when you're stuck or suggest additional test cases, but I need to see more effort coming from you before it gets to the point where you ask for help.

**Comment by piyush0411** [comment]
@lydia-duncan  - Sorry for the way I am approaching. From now on, I shall dig more deep into the files and understand them in a better way. 
In the meantime, could you have a look to the test for this feature I have recently pushed. I wasn't sure for the name of the test file, so if they need to be changed, do let me know.

Thanks!!

**Comment by cassella** [comment]
@piyush0411 - Hi. I'd be surprised if the fix were this simple, since `var x = w / 2` should still have the rhs printed with `printingType == false`.  Have you run the existing chpldoc tests, e.g., `start_test test/chpldoc`?  As in the list append fix, you should run the tests for the feature you're modifying as part of opening a PR.

File: test/chpldoc/functions/functionTypeexprFormat.doc.chpl [hunk 1]
```diff
@@ -0,0 +1,6 @@
+type t = 2*int;
```
Comment (lydia-duncan) [inline test/chpldoc/functions/functionTypeexprFormat.doc.chpl:1]: I'd name this "typeLooksLikeFunc" instead, I think it would be a little more clear to other people.  I also would file this under `test/chpldoc/globals` instead, since these are variable declarations rather than functions

Comment (piyush0411) [reply test/chpldoc/functions/functionTypeexprFormat.doc.chpl:1]: Ok, I shall make the changes.
I am still working on this issue and will let you know as I make some progress.

**Comment by piyush0411** [comment]
@lydia-duncan @cassella 
I have thought of a way to sort this out.
What if we implement a new function of the form 
```
void AstToText::appendExpr(DefExpr* expr, bool printingType, bool isType)
```
This function will be solely for the `type` expr, so we can use it completely for the `type` expr.
I have tested this implementation with all the tests in `chpldoc` folder by running `start_test test/chpldoc/` and the results are as expected. 

> var x = w / 2 should still have the rhs printed with printingType == false

This implementation solves this case too.

Also a small doubt, should `extern type myFloats = 2*real(64);` produce `.. type:: type myFloats = 2*(real(64))` or 
`.. type:: type myFloats = 2 * real(64)`
I think it should be the former one.

Do let me know what you think of this.

**Comment by piyush0411** [comment]
@lydia-duncan 
I guess I have made some blunder here as I tried merging the remote branch. Shall I close this PR and make a new one with the specific changes?

File: compiler/AST/AstToText.cpp [hunk 2]
```diff
@@ -1607,21 +1607,31 @@ void AstToText::appendExpr(CallExpr* expr, bool printingType, const char *outer,
   }
 }
 
+void AstToText::appendExpr(DefExpr* expr, bool printingType, bool isType)
```
Comment (lydia-duncan) [inline compiler/AST/AstToText.cpp:1610]: You never check the value of `printingType` or `isType` in this function.  There isn't a point to sending an argument to a function if it's not going to be used in the function - if you expect it to only be called in certain situations then the function needs to include that in its name so that no one calls your function with isType = false, expecting different behavior

**Comment by lydia-duncan** [comment]
> Also a small doubt, should `extern type myFloats = 2*real(64);` produce `.. type:: type myFloats = 2*(real(64))` or
> `.. type:: type myFloats = 2 * real(64)`
> I think it should be the former one.

I think it should be the latter.  The additional parentheses do not add any value - the parentheses around 64 mean it is always associated with `real` rather than the full expression.  Only `(2*real)(64)` would change that, and that syntax doesn't have meaning in the language today

**Comment by piyush0411** [comment]
> > Also a small doubt, should `extern type myFloats = 2*real(64);` produce `.. type:: type myFloats = 2*(real(64))` or
> > `.. type:: type myFloats = 2 * real(64)`
> > I think it should be the former one.
> 
> I think it should be the latter. The additional parentheses do not add any value - the parentheses around 64 mean it is always associated with `real` rather than the full expression. Only `(2*real)(64)` would change that, and that syntax doesn't have meaning in the language today

For experimentation, I tried the line `var myFloats: 2*real(64);` which produced the output `.. data:: var myFloats: 2*(real(64))`. So shouldn't the two produce same output?

File: compiler/AST/AstToText.cpp (hunk 2 above)
Comment (piyush0411) [reply compiler/AST/AstToText.cpp:1610]: I guess I forgot to include that part in the code. I actually thought of first checking for `isType` and then implementing the body, something like
```
void AstToText::appendExpr(DefExpr* expr, bool printingType, bool isType)
{
  if (isType) {
    mText += expr->sym->name;
    if(expr->init) {
      mText += " = ";
      appendExpr(expr->init, printingType);
    }
  }
}
```

Does this look good?

**Comment by cassella** [comment]
`AppendExpr(CallExpr, ...` intercepts `printingType==true` binary `*` in order to report homogeneous tuples as `n*(type)` instead of as a regular binary operator `n * type`.  (master AstToText.cpp:1388).  If the parentheses around the type aren't desired, the code could be simplified by removing that case, though that seems separate from the purpose of this PR.  That case predates the changes I made in the last few months.

**Comment by lydia-duncan** [comment]
> That case predates the changes I made in the last few months.

Alright, then let's not worry about it right now.  As long as it's consistent, that's what matters.

File: compiler/AST/AstToText.cpp (hunk 2 above)
Comment (lydia-duncan) [reply compiler/AST/AstToText.cpp:1610]: I think it would be better to just have it be a helper function with a different name, like `appendExprTypeVar`.  Have that function only be called when `var->isType()` is true, since you're already performing the check.  Don't take in the printingType argument either, just pass true to the appendExpr call like you have on line 1616 on this branch today

**Comment by piyush0411** [comment]
@lydia-duncan - I have made the changes as you suggested. Please have a look.

Thanks :)

File: compiler/AST/AstToText.cpp [hunk 3]
```diff
@@ -1607,21 +1607,32 @@ void AstToText::appendExpr(CallExpr* expr, bool printingType, const char *outer,
   }
 }
 
+void AstToText::appendExprTypeVar(DefExpr* expr, bool isType)
+{
+  if (isType) {
```
Comment (lydia-duncan) [inline compiler/AST/AstToText.cpp:1612]: This check is a duplicate of the check prior to calling the function.  I don't think it or the isType argument are needed

Comment (piyush0411) [reply compiler/AST/AstToText.cpp]: I have modified the function as you suggested.

**Review by lydia-duncan — APPROVED** [review]
I think this looks right, running testing now.

**Comment by lydia-duncan** [comment]
Nice, this passed a full paratest and everything looks good in a `make docs` run
